src/machine/board_pyportal.go

```diff
@@ -0,0 +1,143 @@
+// +build sam,atsamd51,pyportal
+
+package machine
+
+import "device/sam"
+
+// used to reset into bootloader
+const RESET_MAGIC_VALUE = 0xf01669ef
+
+// GPIO Pins
+const (
+	D0  = PB13 // NINA_RX
+	D1  = PB12 // NINA_TX
+	D2  = PB22 // built-in neopixel
+	D3  = PA04 // PWM available
+	D4  = PA05 // PWM available
+	D5  = PB16 // NINA_ACK
+	D6  = PB15 // NINA_GPIO0
+	D7  = PB17 // NINA_RESETN
+	D8  = PB14 // NINA_CS
+	D9  = PB04 // TFT_RD
+	D10 = PB05 // TFT_RS
+	D11 = PB06 // TFT_CS
+	D12 = PB07 // TFT_TE
+	D13 = PB23 // built-in LED
+	D24 = PA00 // TFT_RESET
+	D25 = PB31 // TFT_WR
+	D32 = PB30 // SD_CS
+	D33 = PA01 // SD_CARD_DETECT
+	D34 = PA16 // LCD_DATA0
+	D35 = PA17 // LCD_DATA1
+	D36 = PA18 // LCD_DATA2
+	D37 = PA19 // LCD_DATA3
+	D38 = PA20 // LCD_DATA4
+	D39 = PA21 // LCD_DATA5
+	D40 = PA22 // LCD_DATA6
+	D41 = PA23 // LCD_DATA7
+	D50 = PA02 // speaker amplifier shutdown
+	D51 = PA15 // NINA_RTS
+
+	NINA_CS     = D8
+	NINA_ACK    = D5
+	NINA_GPIO0  = D6
+	NINA_RESETN = D7
+
+	NINA_TX  = D1
+	NINA_RX  = D0
+	NINA_RTS = D51
+
+	TFT_WR        = PB09
+	TFT_RD        = D9
+	TFT_RS        = D10
+	TFT_CS        = D11
+	TFT_TE        = D12
+	TFT_RESET     = D24
+	TFT_BACKLIGHT = D25
+
+	NEOPIXEL = D2
+	SPK_SD   = D50
+)
+
+// Analog pins
+const (
+	A0 = PA02 // ADC/AIN[0]
+	A1 = D3   // ADC/AIN[2]
+	A2 = PA07 // ADC/AIN[3]
+	A3 = D4   // ADC/AIN[4]
+	A4 = PB00 // ADC/AIN[5]
+	A5 = PB01 // ADC/AIN[10]
+	A6 = PA06 // ADC/AIN[5]
+	A7 = PB08 // ADC/AIN[5]
+
+	AUDIO_OUT = A0
+	LIGHT     = A2
+	TOUCH_YD  = A4
+	TOUCH_XL  = A5
+	TOUCH_YU  = A6
+	TOUCH_XR  = A7
+)
+
+const (
+	LED = D13
+)
+
+// UART0 aka USBCDC pins
+const (
+	USBCDC_DM_PIN = PA24
+	USBCDC_DP_PIN = PA25
+)
+
+// UART1 pins
+const (
+	UART_TX_PIN = D5
+	UART_RX_PIN = D7
+)
+
+/*
+// UART2 pins
+const (
+	UART2_TX_PIN = A4
+	UART2_RX_PIN = A5
+)
+*/
```

Comment:
Why are these left commented out?

Reply:
Right now the UART definitions in machine_atsamd51.go are pinned to specific SERCOM's: https://github.com/tinygo-org/tinygo/blob/master/src/machine/machine_atsamd51.go#L470

It looks like machine_atsamd21.go has a more flexible way of handling this, not sure if something similar could also apply to atsamd51... in any case I thought it might be best to comment this out temporarily while I try to get the other peripherals working.

Reply:
Yes I've been working on making atsamd51 pin configuration simpler (auto-detecting pads and pin modes from the given SERCOM and pin numbers). It'll take a while so maybe it's best to leave UART2 out until that is done.
@deadprogram any opinion on that? I generally prefer no code over commented out potentially useful code.

Reply:
If the code that is commented was the result of a lot of research, and is going to be implemented next, then can leave in comments. Otherwise, agreed.

Reply:
Sounds good to me I'll go ahead and remove any commented out code in my next commit and if necessary replace with more meaningful and specific comments as a reminder.